crates/hir_ty/src/lib.rs

```diff
@@ -552,13 +561,14 @@ mod tests {
             match stmt {
                 hir::Stmt::Let {
                     name,
-                    mutable,
+                    binding,
                     value,
                 } => {
                     let indent = indent(nesting);
                     let name = name.text(self.db);
                     let expr_str = self.debug_expr(hir_file, function, *value, nesting);
-                    let mut_text = hir::testing::format_mutable(*mutable);
+                    let binding = binding.lookup(hir_file.db(self.db));
+                    let mut_text = hir::testing::format_mutable(binding.mutable);
```

Comment:
**Performance Issue**: The new handling for 'binding' introduces additional database lookups which can degrade performance. <br> **Fix**: Cache the results of database lookups where possible to avoid repeated queries and improve performance. <br> **Code Suggestion**: 
 ```
 + let binding_cache = binding.lookup(hir_file.db(self.db));
 + let mut_text = hir::testing::format_mutable(binding_cache.mutable);
 ```